src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1162,6 +1251,25 @@ def _set_thresholds_and_condition(
             else getattr(generation_config, "condition_on_prev_tokens", None)
         )
 
+    @staticmethod
+    def _set_prompt_condition_type(generation_config, prompt_condition_type):
+        allowed_cond_types = ["first-segment", "all-segments"]
```

Comment:
We have to types of prompt_conditioning for long-form:
- 1. We condition only the first segment on the prompt
- 2. We condition all segments on the prompt

Reply:
The original OAI whisper codebase only has i.

Reply:
Looks good!